dotcom-rendering/src/web/components/RewrappedComponent.tsx

```diff
@@ -2,30 +2,32 @@
 import { jsx as _jsx } from 'react/jsx-runtime';
 import { ClassNames } from '@emotion/react';
 
+import type { HTMLTag } from 'src/model/unwrapHtml';
 import { unescapeData } from '../../lib/escapeData';
 
-// Ideally we want to want to avoid an unnecessary 'span' wrapper,
-// which also will cause issues with ads (spacefinder rules). React
-// requires a wrapping element for dangerouslySetInnerHTML so we
-// strip the original element markup and rewrap. The fallback case is
-// added for safety but should never happen. If it does happen
-// though it will cause issues with commercial JS which expects
-// paras to be top-level.
-
+/**
+ * React requires a wrapping element for `dangerouslySetInnerHTML` so we
+ * strip the original element markup and rewrap.
+ *
+ * The fallback case (`isUnwrapped == false`) is added for safety,
+ * but should _never_ happen. We want to avoid an unnecessary `<span>` wrapper,
+ * which can cause issues with ads and commercial JS (`spacefinder` rules).
+ * Commercial expects `<p>` tags at the top level of the rewrapped element.
+ */
```

Comment:
I think that if there case, it should be capture somehow. If we have no visibility on it, how can we hope to fix it?